docs/content/intro/tutorial-preview-environments.md

```diff
@@ -0,0 +1,424 @@
+    BRIGADE_NAMESPACE,
+    true,
+    undefined,
+    undefined,
+    undefined,
+    "type=preview-environment-config",
+  );
+  if (!environmentConfigMaps.body.items.length) {
+    throw Error("No environment configMaps found");
+  }
+  for (const configMap of environmentConfigMaps.body.items) {
+    const projects = yaml.safeLoad(configMap.data.projects);
+    const config = projects[PROJECT_NAME];
+    if (config && config.tag === tag) {
+      const { environmentName } = configMap.metadata.labels;
+      const gitSha = await getTagCommit(tag, config.org, config.repo);
+      await deploy(environmentName, gitSha);
+    }
+  }
+};
+```
+- Our docker images are tagged with git sha so we're calling GitHub API to get it for 
+  current tag.
+
+```js
+const getTagCommit = async (tag, org, repo) => {
+  console.log(`getting commit sha for tag ${tag}`);
+  const tagUrl = `${GITHUB_API_URL}/${org}/${repo}/git/refs/tags/${tag}`;
+  const response = await fetch(tagUrl, {
+    method: "GET",
+    headers: {
+      "Content-Type": "application/json",
+      Authorization: `token ${process.env.BRIGADE_REPO_AUTH_TOKEN}`,
+    },
+  });
+  if (response.ok) {
+    const commit = await response.json();
+    return commit.object.sha;
+  }
+  throw Error(await response.text());
+};
+```
+
+- We execute Brigade Job that runs Helm deployment by providing appropriate docker image tag
+
+```js
+const deploy = async (environmentName, gitSha) => {
+  console.log("deploying helm charts");
+  const service = new Job(
+    "brigade-tutorial-app",
+    "lachlanevenson/k8s-helm:v2.12.3",
+  );
+  service.storage.enabled = false;
+  service.imageForcePull = true;
+  service.tasks = [
+    "cd /src",
+    `helm upgrade ${environmentName}-products \
+    charts/products --install \
+    --namespace=${environmentName} \
+    --set image.tag=${gitSha} \
+    --set replicaCount=1`,
+  ];
+  await service.run();
+};
+```
+
+You can see final implementation of the script in the reference repository:
+
+https://github.com/brigadecore/brigade-tutorial-app/blob/master/brigade.js
+
+## Release
+
+To release our application we follow these steps:
+
+- Commit our changes to GitHub
+- Build and push Docker image
+- Add a `prod` tag to the head of our repository to trigger release
+
+Once `prod` tag is added, GitHub will send payload event to our Brigade GitHub Gateway which will start our release pipeline.
+
+## Summary
+
+In this tutorial we've learnt how to orchestrate creation of isolated Preview Environments where changes can be pushed continuously after every code commit. You can use similar approach for releases to your long lived environments like staging or production. Head over to [brigade-tutorial-config](https://github.com/brigadecore/brigade-tutorial-config) and [brigade-tutorial-app](https://github.com/brigadecore/brigade-tutorial-app) repositories to see complete implementation of brigade scripts used in this tutorial.
```

Comment:
Grammatical/Punctuation: `You can use a similar approach for releases to your long-lived environments, like staging or production.`

Reply:
Referenced repositories appear to be pending creation...